pulsar-client/src/main/java/org/apache/pulsar/client/impl/HttpLookupService.java

```diff
@@ -155,6 +156,30 @@ public String getServiceUrl() {
         return future;
     }
 
+    @Override
+    public CompletableFuture<Optional<SchemaInfo>> getSchema(TopicName topicName, byte[] version) {
+        CompletableFuture<Optional<SchemaInfo>> future = new CompletableFuture<>();
+
+        String schemaName = topicName.getSchemaName();
+        String path = String.format("admin/v2/schemas/%s/schema/%s",
+                schemaName,
+                new String(version, StandardCharsets.UTF_8));
+
+        httpClient.get(path, GetSchemaResponse.class).thenAccept(response -> {
+            future.complete(Optional.of(SchemaInfoUtil.newSchemaInfo(schemaName, response)));
+        }).exceptionally(ex -> {
+            if (ex.getCause() instanceof NotFoundException) {
+                future.complete(Optional.empty());
+            } else {
+                log.warn("Failed to get schema for topic {} version {} : {}",
+                        topicName, version, ex.getCause().getClass());
```

Comment:
I would suggest changing the logging statement to following. because `ex.getCause` can return null.

```
log.warn("Failed to get schema for topic {} version {}", topicName, Base64.encode(version), ex.getCause());
```